brion/plugin/morphologyHDF5.cpp

```diff
@@ -43,320 +43,351 @@ lunchbox::PluginRegisterer<MorphologyHDF5> registerer;
 
 struct Loader
 {
-    MorphologyPlugin& morphology;
-    MorphologyInitData& initData;
+    Loader(MorphologyPlugin& m)
+        : _morphology(m)
+        , _initData(m.getInitData())
+        , _stage("repaired")
+    {
+        lunchbox::ScopedWrite mutex(detail::hdf5Lock());
+        const std::string path = _initData.getURI().getPath();
 
-    H5::H5File file;
+        try
+        {
+            detail::SilenceHDF5 silence;
+            _file.reset(new HighFive::File(path, HighFive::File::ReadOnly));
+        }
+        catch (const HighFive::FileException& exc)
+        {
+            LBTHROW(std::runtime_error(_write
+                                           ? "Could not create morphology file "
+                                           : "Could not open morphology file " +
+                                                 path + ": " + exc.what()));
+        }
+        _checkVersion(path);
+        _selectRepairStage();
+        _readPoints();
+        _readSections();
+        _readSectionTypes();
+        _readPerimeters();
+    }
 
-    H5::DataSet pointDS;
-    hsize_t pointsDims[2];
+    ~Loader()
+    {
+        lunchbox::ScopedWrite mutex(detail::hdf5Lock());
+        _points.reset();
+        _sections.reset();
+        _file.reset();
+    }
 
-    H5::DataSet sectionsDS;
-    hsize_t sectionsDims[2];
+private:
```

Comment:
these reordering seem so subject and are annoying while reviewing. Why do have in a private impl struct things to be private? *sigh*

Reply:
I know it's subjective but it looks more consistent to me this way. As for the reordering I was aware of the potential annoyance, but I decided to go ahead anyway.